cmd/logmonitor/db.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/trillian/types"
+	"github.com/jinzhu/gorm"
+
+	// Include all currently supported gorm dialects
+	_ "github.com/jinzhu/gorm/dialects/mssql"
+	_ "github.com/jinzhu/gorm/dialects/mysql"
+	_ "github.com/jinzhu/gorm/dialects/postgres"
+	_ "github.com/jinzhu/gorm/dialects/sqlite"
+)
+
+// initialEmptyRoot represents an empty trusted root and is used when no
+// information is known about the log being monitored.
+var initialEmptyRoot TrustedLogRoot
+
+func init() {
+	// Set up initialEmptyRoot, which will get used when the database does not
+	// have an existing trusted log root (ie. the first time a tree is
+	// monitored).
+	emptyRoot := &types.LogRootV1{
+		TreeSize: 0,
+	}
+	marshalled, err := emptyRoot.MarshalBinary()
+	if err != nil {
+		glog.Exitf("Could not marshal the empty root: %v\n", err)
+	}
+	initialEmptyRoot = TrustedLogRoot{MarshalledRoot: marshalled, TreeSize: 0}
+}
+
+// TrustedLogRoot is the database model used for storing information about
+// trusted log roots.
+type TrustedLogRoot struct {
```

Comment:
A trusted Log Root is one that has been verified and it's fields have been split out. 
This currently looks like an unverified log root. Consider storing the split out fields from LogRootV1?